### PR TITLE
cdrom: fix get cdrom by serial num. failed issue

### PR DESCRIPTION
**Pull request by xutian**
testing cdrom path get from proc-fs via serial number in this test.
But in fact, cdrom path in proc filesystem not always create
immediately. so try to get cdrom proc-fs path in 90s.

Signed-off-by: Xu Tian xutian@redhat.com

### CODE REVIEW

**Comment by xutian** [comment]
ID: 1297714

File: qemu/tests/cdrom.py [hunk 1]
```diff
@@ -349,7 +349,7 @@ def get_device(vm, dev_file_path):
                 raise error.TestFail("Could not find a valid cdrom device")
         return device
 
-    def get_match_cdrom(session, serial_num):
+    def get_match_cdrom(vm, session, serial_num):
```
Comment (spcui) [inline qemu/tests/cdrom.py]: @xutian please update the docs for param seesion, others look good for me.

File: qemu/tests/cdrom.py [hunk 2]
```diff
@@ -359,21 +359,34 @@ def get_match_cdrom(session, serial_num):
...
+            :param session: VM session
+            :param serial_num: cdrom serial number
+            """
+            show_serial_num = "ls -l /dev/disk/by-id"
+            serial_num_output = session.cmd_output(show_serial_num)
+            if serial_num_output:
+                for line in serial_num_output.splitlines():
+                    if utils_misc.find_substring(str(line), str(serial_num)):
+                        serial_cdrom = line.split(" ")[-1].split("/")[-1]
+                        return serial_cdrom
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: For this func(), can use the https://github.com/avocado-framework/avocado-vt/pull/324 directly now.

File: qemu/tests/cdrom.py [hunk 3]
```diff
@@ -359,21 +359,34 @@ def get_match_cdrom(session, serial_num):
         :return match_cdrom: the cdrom in guest which is corresponding
                              with the CML according to the serial number.
         """
+
+        def get_cdrom_by_serial(session, serial_num):
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: For this func(), can use the avocado-framework/avocado-vt#324 directly now.

**Comment by xutian** [comment]
Hi @CongSmile  thanks your comments, PR updated. would you like to help review. 
thanks Xu

File: qemu/tests/cdrom.py [hunk 4]
```diff
@@ -359,21 +363,26 @@ def get_match_cdrom(session, serial_num):
...
-                                     "serial number %s is same in Qemu"
-                                     " CML.\n Qtree info: %s" %
-                                     (qtree_info, serial_num))
+        serial_cdrom = get_cdrom_by_serial(session, serial_num)
+        if not serial_cdrom:
+            qtree_info = vm.monitor.info("qtree")
+            raise error.TestFail("Could not find the device whose "
+                                 "serial number %s is same in Qemu"
+                                 " CML.\n Qtree info: %s" %
+                                 (qtree_info, serial_num))
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: Should be serial_num, then qtree_info, wrong order.

Comment (CongSmile) [inline qemu/tests/cdrom.py]: And could you please add some debug logs at error ?
            raise error.TestFail("Could not find the corresponding cdrom"
                                 "in guest which is same in Qemu CML.")

**Comment by xutian** [comment]
@CongSmile  really updated, please help review again.

**Comment by suqinhuang** [comment]
LGTM
Acked-by: Suqin Huang

**Comment by suqinhuang** [comment]
Travis CI build failed, others LGTM

File: qemu/tests/cdrom.py [hunk 5]
```diff
@@ -633,6 +633,10 @@ def test(self):
             serial_num = generate_serial_num()
             cdrom = params.get("cdroms", "").split()[-1]
             params["drive_serial_%s" % cdrom] = serial_num
+            cd_format = params.object_params(cdrom).get("cd_format")
+            if cd_format != "virtio_blk":
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: Should be ide, right?

File: qemu/tests/cdrom.py [hunk 6]
```diff
@@ -770,6 +780,10 @@ def test(self):
                 serial_num = generate_serial_num()
                 cdrom = params.get("cdroms", "").split()[-1]
                 params["drive_serial_%s" % cdrom] = serial_num
+                cd_format = params.object_params(cdrom).get("cd_format")
+                if cd_format != "virtio_blk":
+                    extra_param = "wwn=%s" % serial_num
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: Ide?

File: qemu/tests/cdrom.py [hunk 7]
```diff
@@ -795,6 +809,10 @@ def test(self):
                 serial_num = generate_serial_num()
                 cdrom = params.get("cdroms", "").split()[-1]
                 params["drive_serial_%s" % cdrom] = serial_num
+                cd_format = params.object_params(cdrom).get("cd_format")
+                if cd_format != "virtio_blk":
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: ide?

File: qemu/tests/cdrom.py [hunk 8]
```diff
@@ -878,6 +896,10 @@ def test(self):
...
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: ide?

File: qemu/tests/cdrom.py [hunk 9]
```diff
@@ -933,6 +956,10 @@ def test(self):
...
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: ide?

File: qemu/tests/cdrom.py (hunk 5 above)
Comment (xutian) [inline qemu/tests/cdrom.py]: Only virtio_blk not support wwn, so here should be cdrom != "virtio_blk" then add blk_extra_params to device.

File: qemu/tests/cdrom.py (hunk 6 above)
Comment (xutian) [inline qemu/tests/cdrom.py]: same as above. @CongSmile  any other question?

File: qemu/tests/cdrom.py (hunk 5 above)
Comment (CongSmile) [inline qemu/tests/cdrom.py]: could you please also refer 1285256 ?

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by spcui** [comment]
LGTM
Acked-by: scui@redhat.com

**Comment by CongSmile** [comment]
just for a reminder, could you please also refer 1285256 ?

**Comment by spcui** [comment]
@xutian have conflicts, pls update it.

**Comment by suqinhuang** [comment]
Let's ignore ide

**Pull request closed without merging**